### ISSUE TITLE
HLS/M3U8 not working.

### ISSUE THREAD

**Issue by kane2931** [pranav-prakash/RokuCast#27]
Hi, 

Thank you so much for this Chrome Extension.  There's a problem detecting hls/m3u8 links from pages, it always detect as an mp4, resulting when you click download (normaly you should get the .m3u8 text file) but instead you get a file not found error.

Can be reproduced:
Using Clappr (html 5 player) demo page: http://clappr.io/demo/
And with this m3u8 from WGN9 (Chicago): http://wgntribune-lh.akamaihd.net/i/WGNPrimary_1@304622/master.m3u8

Thanks.

**Comment by pranav-prakash** [comment]
Hmm in your example the clappr player doesn't embed the m3u8 url directly in the <video> element (which makes sense since I believe only Apple's safari supports inline hls playback via the <video> element)— instead it uses an indirect method of playback using localstorage and media blob objects, making it hard to find the url via the DOM only. In fact almost all livestreams you come across online will either use this method or their own flash player.

One approach to fixing and finding urls even in livestream situations would be to use chrome extensions' ability to monitor web requests and detect it from there. Though I don't have any plans to redo the extension to find urls via network requests instead of DOM scraping, I'll keep the issue open since it is a valid point and would definitely improve usability.  Another option that requires more manual effort on the part of the user might be to add a button where one can enter the url to cast.

For your situation in particular, you may consider using "channel pear" which I believe allows you to manually enter URLs that will be displayed on the roku. Another option (assuming you want to watch that WGN9 stream frequently) would be to build a simple channel using the roku template.

**Comment by kane2931** [comment]
Yeah an option to enter a custom URL will be great (if it's possible).  Thanks for your quick reply.


On Tuesday, August 1, 2017, 12:31:35 AM GMT-4, Pranav Prakash <notifications@github.com> wrote:


Hmm in your example the clappr player doesn't embed the m3u8 url directly in the element (which makes sense since I believe only Apple's safari supports inline hls playback via the element)— instead it uses an indirect method of playback using localstorage and media blob objects, making it hard to find the url via the DOM only. In fact almost all livestreams you come across online will either use this method or their own flash player.

One approach to fixing and finding urls even in livestream situations would be to use chrome extensions' ability to monitor web requests and detect it from there. Though I don't have any plans to redo the extension to find urls via network requests instead of DOM scraping, I'll keep the issue open since it is a valid point and would definitely improve usability. Another option that requires more manual effort on the part of the user might be to add a button where one can enter the url to cast.

For your situation in particular, you may consider using "channel pear" which I believe allows you to manually enter URLs that will be displayed on the roku. Another option (assuming you want to watch that WGN9 stream frequently) would be to build a simple channel using the roku template.
